find_xml_file.rb

```diff
@@ -0,0 +1,15 @@
+
+def find_xml_file(folder)
+  files = Dir.entries(folder)
+
+  xml_file = nil
+
+  files.each do |filename|
+    if filename.match "\.xml$"
+      xml_file = filename
+      break
+    end
```

Comment:
This will iterate through "files", break and return the first match and not all matches, is that your intention?